src/ocean/util/log/Stats.d

```diff
@@ -650,3 +656,369 @@ unittest
         }
     }
 }
+
+
+/**************************************************************************
+
+    Struct that parses a stats line
+
+**************************************************************************/
+
+public struct StatsLine
```

Comment:
`StatsLine` could just hold line as `cstring` and just slice the `line` passed in `opCall`. That would mean that the lifetime of `StatsLine` is valid as long as `line` passed to `opCall` is valid, but I think that is fine. AFAICS, you don't need to pass the `StatsLine` around, and if you want you could provide `dup` method which would return you a copy of `StatsLine` which would manage its own memory.

That way I think you've removed all allocations here.